### PR TITLE
Merge bitcoin#17251,17250,17279,17299,17306,15756,17203,17719

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
@UdjinM6

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta @UdjinM6 please review

**Comment by PastaPastaPasta** [comment]
maybe make this a few smaller PRs?

**Comment by vijaydasmp** [comment]
> maybe make this a few smaller PRs?

Sure

File: src/bench/bench_dash.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ int main(int argc, char** argv)
     SetupBenchArgs();
     std::string error;
     if (!gArgs.ParseParameters(argc, argv, error)) {
-        tfm::format(std::cerr, "Error parsing command line arguments: %s\n", error.c_str());
+        tfm::format(std::cerr, "Error parsing command line arguments: %s\n", error);
```
Comment (UdjinM6) [inline src/bench/bench_dash.cpp:48]: 17299 / 9adca5a4f9b1d02b825a11930d5f7a5862cd65be : `c_str` changes were squashed into the wrong commit it seems, should go to 17279 / f0f803bab26fc7776688e0fd3bec21ac9aab3b29 (same for similar cases below in this commit)

File: src/wallet/init.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ void WalletInit::AddWalletOptions() const
     gArgs.AddArg("-createwalletbackups=<n>", strprintf("Number of automatic wallet backups (default: %u)", nWalletBackups), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
     gArgs.AddArg("-disablewallet", "Do not load the wallet and disable wallet RPC calls", ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
     gArgs.AddArg("-instantsendnotify=<cmd>", "Execute command when a wallet InstantSend transaction is successfully locked (%s in cmd is replaced by TxID)", ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
-    gArgs.AddArg("-keypool=<n>", strprintf("Set key pool size to <n> (default: %u)", DEFAULT_KEYPOOL_SIZE), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
+    gArgs.AddArg("-keypool=<n>", strprintf("Set key pool size to <n> (default: %u). Warning: Smaller sizes may increase the risk of losing funds when restoring from an old backup, if none of the addresses in the original keypool have been used.", DEFAULT_KEYPOOL_SIZE), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
```
Comment (UdjinM6) [inline src/wallet/init.cpp:49]: 17719: is partial and requires (at least) 15777 and 17260, I'd postpone it for now

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/init.cpp (hunk 2 above)
Comment (vijaydasmp) [reply src/wallet/init.cpp]: ok , dropped 17719 from this PR

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 @PastaPastaPasta please review

File: src/qt/bitcoingui.cpp [hunk 3]
```diff
@@ -536,12 +536,6 @@ void BitcoinGUI::createActions()
         });
     }
 #endif // ENABLE_WALLET
-
-    connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_I), this), &QShortcut::activated, this, &BitcoinGUI::showInfo);
-    connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_C), this), &QShortcut::activated, this, &BitcoinGUI::showConsole);
-    connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_G), this), &QShortcut::activated, this, &BitcoinGUI::showGraph);
-    connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_P), this), &QShortcut::activated, this, &BitcoinGUI::showPeers);
-    connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_R), this), &QShortcut::activated, this, &BitcoinGUI::showRepair);
```
Comment (PastaPastaPasta) [inline src/qt/bitcoingui.cpp:544]: 15756 why are you doing this?

File: src/coinjoin/client.cpp [hunk 4]
```diff
@@ -30,7 +30,7 @@ std::map<const std::string, std::shared_ptr<CCoinJoinClientManager>> coinJoinCli
 CCoinJoinClientQueueManager coinJoinClientQueueManager;
 
 
-void CCoinJoinClientQueueManager::ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStream& vRecv, CConnman& connman, bool enable_bip61)
+void CCoinJoinClientQueueManager::ProcessMessage(CNode* pfrom, const std::string& msg_type, CDataStream& vRecv, CConnman& connman, bool enable_bip61)
```
Comment (PastaPastaPasta) [inline src/coinjoin/client.cpp:33]: 18533, this commit needs to be reformatted such that it is actually a scripted diff and such that CI checks the scripted diff.

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/qt/bitcoingui.cpp (hunk 3 above)
Comment (vijaydasmp) [reply src/qt/bitcoingui.cpp:544]: as suggested by @UdjinM6 https://github.com/UdjinM6/dash/commit/422d9d1d6d9332b7be98110c54f316905a97fa92

File: src/coinjoin/client.cpp (hunk 4 above)
Comment (vijaydasmp) [reply src/coinjoin/client.cpp:33]: sure

File: src/qt/bitcoingui.cpp (hunk 3 above)
Comment (UdjinM6) [reply src/qt/bitcoingui.cpp:544]: yes, 15756 replaces the solution we had before

Comment (C3s1r3m) [reply src/qt/bitcoingui.cpp:544]: César De Jesús Castellanos CDC alministrador y propietario de proyecto services Hash tags DOCS does doesn oh Criptomonedas Bitcoin Ethereum dashboard oh Dashcore

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by UdjinM6** [comment]
ping @PastaPastaPasta

**Comment by vijaydasmp** [comment]
ping @PastaPastaPasta

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**